main_test.go

```diff
@@ -0,0 +1,427 @@
+			source:      "github.com/slsa-framework/example-package",
+			pversiontag: pString("v13"),
+			err:         pkg.ErrorMismatchVersionedTag,
+		},
+		{
+			name:        "versioned v15 no match push-v14.2",
+			artifact:    "binary-linux-amd64-push-v14.2",
+			source:      "github.com/slsa-framework/example-package",
+			pversiontag: pString("v15"),
+			err:         pkg.ErrorMismatchVersionedTag,
+		},
+		{
+			name:        "versioned v15.1 no match push-v14.2",
+			artifact:    "binary-linux-amd64-push-v14.2",
+			source:      "github.com/slsa-framework/example-package",
+			pversiontag: pString("v15.1"),
+			err:         pkg.ErrorMismatchVersionedTag,
+		},
+		// Special case of the e2e test repository building builder from head.
+		{
+			name:      "e2e test repository verified with builder at head",
+			artifact:  "binary-linux-amd64-e2e-builder-repo",
+			source:    "github.com/slsa-framework/example-package",
+			branch:    "main",
+			noversion: true,
+		},
+		// Malicious builders and workflows.
+		{
+			name:      "rekor upload bypassed",
+			artifact:  "binary-linux-amd64-no-tlog-upload",
+			source:    "github.com/slsa-framework/example-package",
+			err:       pkg.ErrorNoValidRekorEntries,
+			noversion: true,
+		},
+		{
+			name:      "malicious: untrusted builder",
+			artifact:  "binary-linux-amd64-untrusted-builder",
+			source:    "github.com/slsa-framework/example-package",
+			err:       pkg.ErrorUntrustedReusableWorkflow,
+			noversion: true,
+		},
+		{
+			name:      "malicious: invalid signature expired certificate",
+			artifact:  "binary-linux-amd64-expired-cert",
+			source:    "github.com/slsa-framework/example-package",
+			err:       pkg.ErrorNoValidRekorEntries,
+			noversion: true,
+		},
+		// Regression test of sharded UUID
+		{
+			name:      "regression: sharded uuids",
+			artifact:  "binary-linux-amd64-sharded",
+			source:    "github.com/slsa-framework/slsa-verifier",
+			branch:    "release/v1.0",
+			noversion: true,
+		},
+	}
+	for _, tt := range tests {
+		tt := tt // Re-initializing variable so it is not changed while executing the closure below
+		t.Run(tt.name, func(t *testing.T) {
+			// t.Parallel()
+
+			checkVersions := golangGeneratorVersions
+			if tt.noversion {
+				checkVersions = []string{""}
+			}
+
+			for _, v := range checkVersions {
+				branch := tt.branch
+				if branch == "" {
+					branch = "main"
+				}
+
+				artifactPath = filepath.Clean(fmt.Sprintf("./testdata/go/%v/%s", v, tt.artifact))
+				provenancePath = fmt.Sprintf("%s.intoto.jsonl", artifactPath)
+
+				_, err := runVerify("", artifactPath,
+					provenancePath,
+					tt.source, branch,
+					tt.ptag, tt.pversiontag)
+
+				if !errCmp(err, tt.err) {
+					t.Errorf(cmp.Diff(err, tt.err, cmpopts.EquateErrors()))
+				}
+			}
+		})
+	}
+}
+
+func TestContainerVerification(t *testing.T) {
+	runContainerVerification = func(ctx context.Context, path string) ([]oci.Signature, string, error) {
```

Comment:
Where is this defined?

Reply:
A merge problem, defined in container pkg: https://github.com/slsa-framework/slsa-verifier/blob/77f8bc623b96f6124ff2cc4a1daade743f01c6b3/cli/slsa-verifier/main_test.go#L516

FWIW there is an issue filed here to support container type interfaces, so these funcs vars will go away https://github.com/slsa-framework/slsa-verifier/issues/220

Reply:
Closing.